library/cartridge_check_instance_started.py

```diff
@@ -4,32 +4,94 @@
 from ansible.module_utils.helpers import ModuleRes, CartridgeException
 from ansible.module_utils.helpers import get_control_console
 
-
 argument_spec = {
     'control_sock': {'required': True, 'type': 'str'},
     'stateboard': {'required': True, 'type': 'bool'},
+    'expected_states': {'required': False, 'type': 'list', 'default': ['Unconfigured', 'RolesConfigured']},
+    'check_buckets_are_discovered': {'required': False, 'type': 'bool', 'default': False},
 }
 
 
-def check_instance_started(params):
-    try:
-        control_console = get_control_console(params['control_sock'])
-        instance_is_alive = True
-        if not params['stateboard']:
-            instance_is_alive, _ = control_console.eval_res_err('''
-                return require('membership').myself().status == 'alive'
+def check_stateboard_started(control_console):
+    box_status, err = control_console.eval_res_err('''
+        if type(box.cfg) == 'function' then
+            return nil, "Box hasn't been configured"
+        end
+        return box.info.status
+    ''')
+    if err is not None:
+        return ModuleRes(success=False, msg="Impossible to get stateboard status: %s" % err)
+    if box_status != 'running':
+        return ModuleRes(success=False, msg="Stateboard is not running, it has '%s' status" % box_status)
+
+    return ModuleRes(success=True)
+
+
+def check_instance_started(control_console, expected_states, check_buckets_are_discovered):
+    instance_info, err = control_console.eval_res_err('''
+        local myself = require('membership').myself()
+        return { myself.status, myself.payload and myself.payload.state }
```

Comment:
Checking `require('cartridge.confapplier').get_state()` is more reliable. And I think OperationError would also be acceptable for the ansible.